services/src/main/java/org/keycloak/social/instagram/InstagramIdentityProvider.java

```diff
@@ -46,23 +47,20 @@ public InstagramIdentityProvider(KeycloakSession session, OAuth2IdentityProvider
 
 	protected BrokeredIdentityContext doGetFederatedIdentity(String accessToken) {
 		try {
-			JsonNode raw = SimpleHttp.doGet(PROFILE_URL,session).param("access_token", accessToken).asJson();
+			JsonNode raw = SimpleHttp.doGet(PROFILE_URL,session)
+				.param("access_token", accessToken)
+				.param("fields", PROFILE_FIELDS)
+				.asJson();
 			
 			JsonNode profile = raw.get("data");
 			
 			logger.debug(profile.toString());
 
 			String id = getJsonProperty(profile, "id");
+	  		String username = getJsonProperty(profile, "username");
```

Comment:
raw no 'data' of node

```suggestion
			JsonNode raw = SimpleHttp.doGet(PROFILE_URL,session)
				.param("access_token", accessToken)
				.param("fields", PROFILE_FIELDS)
				.asJson();
			
			logger.debug(raw.toString());

			String id = getJsonProperty(raw, "id");
	  		String username = getJsonProperty(raw, "username");
```